rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -1010,7 +1010,7 @@ public synchronized List<InternalService> buildInternalServices() {
 
         if (getRskSystemProperties().isSyncEnabled()) {
             internalServices.add(getSyncPool());
-            if (getSyncConfiguration().isServerSnapSyncEnabled()) {
+            if (getSyncConfiguration().isServerSnapSyncEnabled() || getRskSystemProperties().getSyncMode().equalsIgnoreCase("snap")) {
```

Comment:
I guess we should only add this service only if the server part is enable (as it is now), regardless of client is enabled or not

Reply:
Addressed the issue by reverting the change, the service is now only added if the snap server is enabled